src/subsystems/memory/memory_types.rs

```diff
@@ -0,0 +1,230 @@
+use crate::{crtp_utils::WaitForPacket, Error, Result};
+use crazyflie_link::Packet;
+use flume as channel;
+use std::convert::{TryFrom, TryInto};
+
+use crate::crazyflie::MEMORY_PORT;
+
+const READ_CHANNEL: u8 = 1;
+const WRITE_CHANNEL: u8 = 2;
+
+const MEM_MAX_REQUEST_SIZE: usize = 20;
+
+
+/// Description of a memory in the Crazyflie
+#[derive(Debug, Clone)]
+pub struct MemoryBackend {
+    /// Unique identifier for this memory subsystem (used when reading/writing/querying)
+    pub memory_id: u8,
+    /// Type of memory
+    pub memory_type: MemoryType,
+
+    pub(crate) uplink: channel::Sender<Packet>,
+    //TODO: Lock this!
```

Comment:
Does this imply multiple readers/writers can access this right now?

Reply:
Good catch, fixed!